core/src/main/java/tc/oc/pgm/modules/ModifyBowProjectileModule.java

```diff
@@ -8,39 +8,49 @@
 import org.bukkit.potion.PotionEffect;
 import org.jdom2.Document;
 import org.jdom2.Element;
+import tc.oc.pgm.api.filter.Filter;
 import tc.oc.pgm.api.map.MapModule;
 import tc.oc.pgm.api.map.factory.MapFactory;
 import tc.oc.pgm.api.map.factory.MapModuleFactory;
 import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.filters.parse.FilterParser;
 import tc.oc.pgm.util.xml.InvalidXMLException;
 import tc.oc.pgm.util.xml.XMLUtils;
 
 public class ModifyBowProjectileModule implements MapModule<ModifyBowProjectileMatchModule> {
   protected final Class<? extends Entity> cls;
   protected final float velocityMod;
   protected final Set<PotionEffect> potionEffects;
+  protected final Filter pickupFilter;
 
   public ModifyBowProjectileModule(
-      Class<? extends Entity> cls, float velocityMod, Set<PotionEffect> effects) {
+      Class<? extends Entity> cls,
+      float velocityMod,
+      Set<PotionEffect> effects,
+      Filter pickupFilter) {
     this.cls = cls;
     this.velocityMod = velocityMod;
     potionEffects = effects;
+    this.pickupFilter = pickupFilter;
   }
 
   @Override
   public ModifyBowProjectileMatchModule createMatchModule(Match match) {
     return new ModifyBowProjectileMatchModule(
-        match, this.cls, this.velocityMod, this.potionEffects);
+        match, this.cls, this.velocityMod, this.potionEffects, this.pickupFilter);
   }
 
   public static class Factory implements MapModuleFactory<ModifyBowProjectileModule> {
     @Override
     public ModifyBowProjectileModule parse(MapFactory factory, Logger logger, Document doc)
         throws InvalidXMLException {
+      FilterParser filters = factory.getFilters();
+
       boolean changed = false;
       Class<? extends Entity> projectile = Arrow.class;
       float velocityMod = 1;
       Set<PotionEffect> potionEffects = new HashSet<>();
+      Filter pickupFilter = null;
```

Comment:
```suggestion
      Filter pickupFilter = StaticFilter.ALLOW;
```
Generally filters can just default to allow/deny instead of null which can make things easier as you avoid null-checks elsewhere